### ISSUE TITLE
Question on LossCompute

### ISSUE THREAD

**Issue by IliasMAOUDJ** [marcelomatheusgauy/Audio_mfcc_gram_transformers#1]
Dear Marcelo Matheus Gauy,

I found out you paper about audio mfcc transformers and wanted to apply to another pathological voice detection. I'm a bit confused about the computation of your loss. You compute the cross_entropy_loss between x[:,0:1,1:2] and y. Why do you take only the first frame of x when it is an array of more than a hundred values ? All the other frames are ignored. Your prediction relies only on this first frame or have I misunderstood ? Thanks 

```
class LossCompute:
    "A simple loss compute and train function."
    def __init__(self, generator, softgenerator, opt=None, pretrain=True):
        self.generator = generator
        self.softgenerator = softgenerator
        self.opt = opt
        self.pretrain = pretrain
        
    def __call__(self, x, y, mask_label):
        train_acc = 0
        if self.pretrain:
            x = self.generator(x)
            L1_loss = nn.L1Loss()
            loss = L1_loss(x, y)
        else:
            x = self.softgenerator(x)
            cross_entropy_loss = nn.BCELoss()
            y = y.unsqueeze(1).unsqueeze(1)
            loss = cross_entropy_loss(x[:,0:1,1:2], y)
            
            #compute accuracy
            _, predicted = torch.max(x[:, 0:1], 2)
            train_acc = torch.sum(predicted == y[:, 0])/y.shape[0]
            
            
        if self.opt is not None:
            loss.backward()
            self.opt.step()
            self.opt.optimizer.zero_grad()
        #print(loss.data.shape)
        return loss.data.item(), train_acc
```
